FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -48,9 +54,14 @@ async Task<RequestResult> DoRequestAsync(HttpMethodKind method, string path, str
 
             var startTime = DateTime.UtcNow;
 
-            var message = new HttpRequestMessage(new HttpMethod(method.Name()), path);
+            var message = new HttpRequestMessage(new HttpMethod(method.Name()), $"{endpoint}{path}");
```

Comment:
why did you add `endpoint` here? the client already has the endpoint as the `BaseAddress`

Reply:
If the user is passing their own then they might not have a base address set and you don't want to change theirs or set one

Reply:
can you add a test where you create your own `HttpClient` not specifying `BaseAddress` on it, just to make sure we are handling the `endpoint` + `path` correctly

Reply:
sure what test file do you want it in?

Reply:
ClientTest.cs?

Reply:
How do I run unit tests that require networking? they say connection refused do I need to supply a secret key somewhere?

Reply:
The client should act the same though with user supplied or without because even if the client creates its own its not setting a base. So even without user supplied httpclient if it doesn't work all the current tests will fail

Reply:
pretty much all tests extends this https://github.com/fauna/faunadb-csharp/blob/master/FaunaDB.Client.Test/TestCase.cs#L42 so you can set an env variable with your key/secret and domain/port.

Anyway, without a test I cannot ensure that a custom httpclient set with a BaseAddress will not interfere with endpoint+path

Reply:
Or you can set the base address of the custom httpclient to a non related address and ensure we are doing requests to fauna address specified on endpoint

Reply:
Ok i passed an httpclient in TestCase with a random baseaddress. it worked as expected and the tests passed. HttpRequestMessage will always override the base address if a domain is passed to it